core/src/services/s3/error.rs

```diff
@@ -49,9 +49,10 @@ pub(super) fn parse_error(resp: Response<Buffer>) -> Error {
         _ => (ErrorKind::Unexpected, false),
     };
 
-    let (message, s3_err) = de::from_reader::<_, S3Error>(body.clone().reader())
+    let body_content = body.chunk();
```

Comment:
We can convert `Buffer` to `Bytes` first like we do for azblob: 

```
let bs = body.to_bytes();

Reply:
fixed